polars/polars-io/src/csv/read.rs

```diff
@@ -381,6 +381,11 @@ impl<'a, R: MmapBytesReader + 'a> CsvReader<'a, R> {
         let fields = overwriting_schema.iter_fields().filter_map(|mut fld| {
             use DataType::*;
             match fld.data_type() {
+                Date | Datetime(_, _) => {
```

Comment:
I don't think this is correct. I believe this would lead to the column being read as utf8 and then cast afterwards, which is much more expensive than immediately parsing.

What is the reason the unix milliseconds fail?